pages/claim.js

```diff
@@ -1,9 +1,45 @@
 import React from 'react'
-import Frame from '../components/Frame'
+
+import { compose } from 'react-apollo'
+import { withRouter } from 'next/router'
+import isEmail from 'validator/lib/isEmail'
+
+import { maybeDecode } from '../lib/utils/base64u'
 import ClaimMembership from '../components/Account/Memberships/Claim'
+import Frame from '../components/Frame'
+import withT from '../lib/withT'
+
+const ALLOWED_CONTEXT = ['claim', 'access']
 
-export default () => (
-  <Frame>
-    <ClaimMembership />
+const Claim = ({ router: { query }, t }) => {
+  let { context, email, code } = query
+
+  context = ALLOWED_CONTEXT.includes(context) && context
+  email = email && maybeDecode(email)
+  email = email && isEmail(email) ? email : ''
+  code =
+    code && maybeDecode(code)
+      .substr(0, 10) // Limit code to 10 chars
+      .replace(/[^a-zA-Z0-9]/g, '') // Remove unknown (or unwanted) chars
```

Comment:
Looks like we're doing this twice, why?

Reply:
Indeed. Wanted to make sure `code` provided via query parameters is really, really stripped of anything undesirable. Meanwhile `handleVoucherCode` will also sanitize a user entered voucher code before submitting e.g. `hans peter ` becomes `HANSPETER`.

Wondering if there is way, safe enough to circumvent double-sanitization.

Reply:
Makes sense. Better be safe than sorry.